src/cli/java/org/commcare/util/screen/CompoundScreenHost.java

```diff
@@ -31,8 +31,8 @@ public void prompt(PrintStream out) throws CommCareSessionException {
     }
 
     @Override
-    public final boolean handleInputAndUpdateSession(CommCareSession session, String input) throws CommCareSessionException {
-        if (getCurrentScreen().handleInputAndUpdateHost(input, this)) {
+    public final boolean handleInputAndUpdateSession(CommCareSession session, String input, boolean allowAutoLaunch) throws CommCareSessionException {
```

Comment:
Flags like this always cause me a bit of pause because it's not obvious when and how the autolaunch behavior should be triggered and propagated.

What are the circumstances in which one wouldn't propagate the autolaunch? Is the flag there primarily to hedge against the possibility that it might cause problems in some circumstance?

Reply:
> Is the flag there primarily to hedge against the possibility that it might cause problems in some circumstance?

Originally, yes. That flag exists because at the time of adding it, I wasn't confident enough in formplayer test coverage to be sure it wasn't going to break another workflow. I don't remember if there was actually a known workflow that broke without the flag.

Reply:
I took a paas at refining the `allowAutoLaunch` flag a bit and it seems like the only place it's tunred off now is while [rebuilding the session](https://github.com/dimagi/formplayer/blob/autoLaunchRevamp/src/main/java/org/commcare/formplayer/services/MenuSessionFactory.java#L79) which seems intended as we probably want to bypass the mark and rewind worklfow caused by case search while rebuilding session.